### DISCUSSION TITLE
working with collections per page

### DISCUSSION THREAD

**Discussion by ghost** [tighten/jigsaw#613] [Q&A] [answered]
I apologized for asking a lot of questions regarding jigsaw since I've been playing around with this and building at the same time. I do want to ask is that i don't really see much advanced topics of what jigsaw can do. 

Is there a way for me to customize perPage in other words, I believe by default for posts i believe or any kind of collections is set to 10 posts at the same time. now the main question is question is that if I'm working on a landing page, how do i implement a custom per page to like for example, i want to use per Page and set it to two posts and that should grab the lastest published posts. 

I hope that that make sense.

**Comment by bakerkretzmar** [comment, marked as answer]
Yes, you can customize a collection's `perPage` value to show a number other than 10: https://jigsaw.tighten.com/docs/collections-pagination/

For the situation you're describing though, just showing two recent posts on a home page, it's probably a lot simpler to just 'query' for those two posts directly. Try something like this:

```blade
@foreach ($posts->sortByDesc('published_at')->take(2) as $recentPost)
    {{-- posts here --}}
@endforeach
```

Let me know if that helps!

**Reply by ghost** [reply]
that works! cool stuff! Tho i might have more questions later :D... please marked as answer

**Reply by ghost** [reply]
how do i sort by random

**Reply by bakerkretzmar** [reply]
Try the Collection `->shuffle()` method ([docs](https://laravel.com/docs/9.x/collections#method-shuffle)).

**Reply by ghost** [reply]
I'll try that.. thanks

**Comment by ghost** [comment]
one last thing regarding this, is there also when u use collections->shuffle(), is there a way to not show a certain current page if you are on that page.. in other words. 
```
domain.com/project/silver-quantum
```
and if i create a new recent portfolio section and do the same foreach $portfolios->shuffle()->take(2) but add exclude $current_page.

**Reply by ghost** [reply]
i did find except for collections but not sure how to use it apparently. but i have a collections for portfolios

**Comment by ghost** [comment]
so basically what I'm trying to do is is for example

`https://domain.com/portfolio/silver-quantum` in this specific page, there's a another section called recent works and i have is
```
foreach ($portfolios->shuffle()->take(3) as $portflio ) {

}
```
I want to add except() to exclude the same portfolio (silver quantum) from this recent works since I'm already on that page.

**Reply by bakerkretzmar** [reply]
`except()` filters based on keys I think, so that'll only work for example if your collection is keyed by slug or something. I would just use `filter()` for this, you can pass it a function and do any checks you need inside there.

**Reply by ghost** [reply]
would you be able to show me an example for usig filter for my case. that way it would be easier for me to understand. I think i might need like to check fo the {-title} is the same as the current page then don't display it.

**Reply by bakerkretzmar** [reply]
Maybe something like this?

```php
// $page is the current portfolio
foreach ($portfolios->filter(fn ($portfolio) => $portfolio->slug !== $page->slug)->shuffle()->take(3) as $portfolio) {
    //
}
```

**Reply by ghost** [reply]
that didn't work but this seems to work fine... from what i saw
`<?php foreach ( $portfolios->filter( fn( $portfolio ) => $portfolio->title !== $page->title )->shuffle()->take( 3 ) as $portfolio ) : ?>`

**Reply by ghost** [reply]
thank you, you've been helpful for the last couple of months or so... appreciated.

**Reply by bakerkretzmar** [reply]
You're welcome, glad I can help!
